packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -810,26 +800,16 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, IERC900
 
     /**
     * @dev Internal function to stake an amount of tokens for a guardian
-    * @param _from Address sending the amount of tokens to be deposited
     * @param _guardian Address of the guardian to deposit the tokens to
     * @param _amount Amount of tokens to be deposited
     * @param _data Optional data that can be used to request the activation of the deposited tokens
     */
-    function _stake(address _from, address _guardian, uint256 _amount, bytes memory _data) internal {
+    function _stake(address _guardian, uint256 _amount, bytes memory _data) internal {
         require(_amount > 0, ERROR_INVALID_ZERO_AMOUNT);
         _updateAvailableBalanceOf(_guardian, _amount, true);
 
-        // Activate tokens if it was requested by the sender. Note that there's no need to check
-        // the activation amount since we have just added it to the available balance of the guardian.
-        if (_data.toBytes4() == GuardiansRegistry(this).activate.selector) {
-            _activateTokens(_guardian, _amount, _from);
-        } else if (_data.toBytes4() == GuardiansRegistry(this).lockActivation.selector) {
-            _activateTokens(_guardian, _amount, _from);
-            _lockActivation(_guardian, _from, _amount);
-        }
-
         emit Staked(_guardian, _amount, _totalStakedFor(_guardian), _data);
-        require(guardiansToken.safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
+        require(guardiansToken.safeTransferFrom(msg.sender, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
```

Comment:
⚠️ shouldn't this be `_guardian`?

Reply:
As explained above, these assets should be transferred from the real sender in `stake`